File: EPZEU_PR/property-register/pr-domain/src/main/java/bg/registryagency/epzeu/pr/domain/model/ApplicationProcessContent.java
package bg.registryagency.epzeu.pr.domain.model;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * <div class="bg">Модел на съдържание на заявление.</div>
 */
@Getter
@Setter
@NoArgsConstructor
@EqualsAndHashCode
public class ApplicationProcessContent {
    /** <div class="bg">Идентификатор на съдържание на заявление.</div> */
    private Long applicationProcessContentId;
    /** <div class="bg">Съдържанието на заявлението в битове.</div> */
    private byte[] content;
    /** <div class="bg">
     * Тип на съдържанието:
     *  <li>{@link ApplicationProcessContent.Type#JSON} - съдържанието е в JSON формат, докато заявлението е чернова.</li>
     *  <li>{@link ApplicationProcessContent.Type#XML} - съдържанието е в XML формат, когато заявлението е готово и може да бъде изпратено към бек офис системата.</li>
     * </div>
     */
    private ApplicationProcessContent.Type type;
    /** <div class="bg">Процес на заявяване на услуга, в който участва Съдържанието на заявление.</div> */
    private ApplicationProcess applicationProcess;

    /**
     * <div class="bg">Конструктор за създаване на Съдържание на заявление.</div>
     * @param applicationProcessContentId <div class="bg">Идентификатор на Съдържание на заявление.</div>
     */
    public ApplicationProcessContent(Long applicationProcessContentId) {
        this.applicationProcessContentId = applicationProcessContentId;
    }

    /**
     * <div class="bg">Конструктор за създаване на Съдържание на заявление.</div>
     * @param applicationProcessContentId <div class="bg">Идентификатор на Съдържание на заявление.</div>
     * @param type <div class="bg">Тип на Съдържание на заявлението.</div>
     * @param applicationProcessId <div class="bg">Процес на заявяване на услуга, в който участва Съдържанието на заявлението.</div>
     */
    public ApplicationProcessContent(Long applicationProcessContentId, Type type, Long applicationProcessId) {
        this.applicationProcessContentId = applicationProcessContentId;
        this.type = type;
        this.applicationProcess = new ApplicationProcess(applicationProcessId);
    }

    /**
     * <div class="bg">
     *     Тип на формата на съдържанието на заявление.
     *      <li>{@link #JSON} - съдържанието е в JSON формат, докато е заявлението чернова.</li>
     *      <li>{@link #XML} - съдържанието е в XML формат, когато заявлението е готово и може да бъде изпратено към бек офис системата.</li>
     * </div>
     */
    @Getter
    public enum Type {
        JSON, XML
    }
}

File: EPZEU_CR/.NET/EPZEU.CR/ApplicationProcesses/Models/ApplicationProcessContent.cs
using EPZEU.Utilities;
using System.IO;

namespace EPZEU.CR.ApplicationProcesses.Models
{
    /// <summary>
    /// Тип на данните.
    /// </summary>
    public enum ApplicationProcessContentTypes
    {
        ApplicationJSON = 1,
        PackageXML = 2,
    }

    /// <summary>
    /// Модел на съдържание на пакети за вписване.
    /// </summary>
    public class ApplicationProcessContent
    {
        /// <summary>
        /// Уникален идентификатор на съдържание на пакети за вписване.
        /// </summary>
        [DapperColumn("application_process_content_id")]
        public long? ApplicationProcessContentID { get; set; }

        /// <summary>
        /// Идентификатор на данни за процеси на заявяване на услуга.
        /// </summary>
        [DapperColumn("application_process_id")]
        public long? ApplicationProcessID { get; set; }

        /// <summary>
        /// Тип на данните: 1 - application_json; 2 - package_xml;.
        /// </summary>
        [DapperColumn("type")]
        public ApplicationProcessContentTypes? Type { get; set; }

        /// <summary>
        /// Съдържание.
        /// </summary>
        [DapperColumn("content")]
        public Stream Content { get; set; }
    }
}
